std/string.d

```diff
@@ -2033,6 +2040,224 @@ S[] splitLines(S)(S s, in KeepTerminator keepTerm = KeepTerminator.no) @safe pur
     });
 }
 
+/***********************************
+ *  Split an array or slicable range of characters into a range of lines
+    using $(D '\r'), $(D '\n'),
+    $(D "\r\n"), $(XREF uni, lineSep), and $(XREF uni, paraSep) as delimiters.
+    If $(D keepTerm) is set to $(D KeepTerminator.yes), then the delimiter
+    is included in the slices returned.
+
+    Does not throw on invalid UTF; such is simply passed unchanged
+    to the output.
+
+    Does not allocate memory.
+
+  Params:
+    r = array of $(D chars), $(D wchars), or $(D dchars) or a slicable range
+    keepTerm = whether delimiter is included or not in the results
+  Returns:
+    range of slices of the input range $(D r)
+
+  See_Also:
+    $(LREF splitLines)
+    $(XREF algorithm, splitter)
+    $(XREF regex, splitter)
+ */
+auto splitterLines(KeepTerminator keepTerm = KeepTerminator.no, Range)(Range r)
+if ((hasSlicing!Range && hasLength!Range) ||
+    isSomeString!Range)
+{
+    import std.uni : lineSep, paraSep;
+    import std.conv : unsigned;
+
+    static struct Result
+    {
+    private:
+        Range _input;
+        alias IndexType = typeof(unsigned(_input.length));
```

Comment:
It's murky territory. For example, the `length` property is allowed to always return `ulong` (according to `hasLength`), which breaks code that assumes `size_t` when the target is a 32-bit platform. Yet many range algorithms and higher-order ranges treat the `length` property as being `size_t`. I don't think it's a major problem, but in the long term we need to settle on which way to go.